app/controllers/posts_controller.rb

```diff
@@ -1,14 +1,23 @@
 class PostsController < ApplicationController
   load_and_authorize_resource
-
   def index
     @user = User.find(params[:user_id])
     @posts = @user.posts.includes(:comments)
+    respond_to do |format|
+      format.html
+      format.xml { render xml: @posts }
+      format.json { render json: @posts, status: :ok }
+    end
   end
 
   def show
     @post = Post.find(params[:id])
     @user = User.find(@post.author_id)
+    respond_to do |format|
+      format.html
+      format.xml { render xml: @posts.comments }
+      format.json { render json: @posts.comments }
+    end
```

Comment:
a- [ ] Nice adding JSON format but when I try to access the posts or comments from API app like Thunder or Postman I got the following error because csrf token prevent requests from an unknown source 
I suggest adding URLs for API separated from normal front-end app URLs  and trying to skip verify the authenticity token 
- I added a user , post and a comment 
<img width="1440" alt="Screenshot 2023-05-26 at 18 32 31" src="https://github.com/alicemirigo92/blog_post/assets/61790587/3e8cdff5-bbe8-4d6b-8777-34e5fc30522c">

Reply:
Done, as advised